realis.go

```diff
@@ -62,13 +63,146 @@ type realisClient struct {
 	readonlyClient *aurora.ReadOnlySchedulerClient
 }
 
+type option func(*RealisConfig)
+
+//Config sets for options in RealisConfig.
+func BasicAuth(username, password string) option {
+
+	return func(config *RealisConfig) {
+		config.username = username
+		config.password = password
+	}
+}
+
+func SchedulerUrl(url string) option {
+	return func(config *RealisConfig) {
+		config.url = url
+	}
+}
+
+func TimeoutMS(timeout int) option {
+	return func(config *RealisConfig) {
+		config.timeoutms = timeout
+	}
+}
+
+func ZKCluster(cluster *Cluster) option {
+	return func(config *RealisConfig) {
+		config.cluster = cluster
+	}
+}
+
+func ZKUrl(url string) option {
+	return func(config *RealisConfig) {
+		config.cluster = GetDefaultClusterFromZKUrl(url)
+	}
+}
+
+func Retries(backoff *Backoff) option {
+	return func(config *RealisConfig) {
+		config.backoff = backoff
+	}
+}
+
+func ThriftJSON() option {
+	return func(config *RealisConfig) {
+		config.jsonTransport = true
+	}
+}
+
+func ThriftBinary() option {
+	return func(config *RealisConfig) {
+		config.binTransport = true
+	}
+}
+
+func BackOff(b *Backoff) option {
+	return func(config *RealisConfig) {
+		config.backoff = b
+	}
+}
+
+func NewRealisClient(options ...option) (Realis, error) {
+	config := &RealisConfig{}
+	fmt.Println(" options length: ", options)
+	for _, opt := range options {
+		opt(config)
+	}
+	//Default timeout
+	if config.timeoutms == 0 {
+		config.timeoutms = 10000
+	}
+	//Set default Transport to JSON if needed.
+	if !config.jsonTransport && !config.binTransport {
+		config.jsonTransport = true
+	}
+	var url string
+	var err error
+	//Cluster or URL?
+	if config.cluster != nil {
+		url, err = LeaderFromZK(*config.cluster)
+		if err != nil {
+			fmt.Errorf("LeaderFromZK error: %+v\n ", err)
+		}
+		fmt.Println("schedURLFromZK: ", url)
+	} else if config.url != "" {
+		fmt.Println("Scheduler URL: ", config.url)
+		url = config.url
+	} else {
+		return nil, errors.New("Incomplete Options -- url or cluster required")
+	}
+
+	if config.jsonTransport {
+		trans, err := defaultTTransport(url, config.timeoutms)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error creating realis")
+		}
+		httpTrans := (trans).(*thrift.THttpClient)
+		httpTrans.SetHeader("Content-Type", "application/x-thrift")
+		config.transport = trans
+		config.protoFactory = thrift.NewTJSONProtocolFactory()
+	} else if config.binTransport {
+		trans, err := defaultTTransport(url, config.timeoutms)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error creating realis")
+		}
+		httpTrans := (trans).(*thrift.THttpClient)
+		httpTrans.SetHeader("Accept", "application/vnd.apache.thrift.binary")
+		httpTrans.SetHeader("Content-Type", "application/vnd.apache.thrift.binary")
+		httpTrans.SetHeader("User-Agent", "GoRealis v1.0.4")
```

Comment:
This should be `httpTrans.SetHeader("User-Agent", "GoRealis v"+VERSION)` so that we don't have to update it everywhere when we change it :slightly_smiling_face: